### PR TITLE
HDDS-10181. Remove duplicate ACL protobuf definitions in interface-storage

### PR DESCRIPTION
**Pull request by ivandika3**
## What changes were proposed in this pull request?

It's observed that the ACL serialization (toProtobuf) and deserialization (fromProtobuf) of OmPrefixInfo uses OzoneAclStorageUtil, whereas OmKeyInfo, OmBucketInfo, OmVolumeArgs uses OzoneAclUtil. The logic for both classes looks the same.

There are also duplicate proto definitions for OzoneAclInfo in OmStorageProtocol.proto and OmClientProtocol.proto.

We need to remove the duplication to prevent any divergence.

Note: This seems the separation was done as part of HDDS-4242, under HDDS-4216 which are supposed to separate RPC and storage proto files for compatibility reasons by introducing interface-storage package. But it seems only PrefixInfo and UserVolumeInfo have been done for this. If these changes are intended, this can be closed and we can instead migrate the other OzoneAcls to use the one in interface-storage. 
## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10181

## How was this patch tested?

Existing tests.

Clean CI: https://github.com/ivandika3/ozone/actions/runs/7611899320

### CODE REVIEW

File: hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/helpers/OmPrefixInfo.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ public static OmPrefixInfo getFromProtobuf(PersistedPrefixInfo prefixInfo) {
           .getFromProtobuf(prefixInfo.getMetadataList()));
     }
     if (prefixInfo.getAclsList() != null) {
-      opib.setAcls(OzoneAclStorageUtil.fromProtobuf(prefixInfo.getAclsList()));
+      opib.setAcls(OzoneAclUtil.fromProtobuf(prefixInfo.getAclsList()));
```
Comment (sumitagrawl) [inline hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/helpers/OmPrefixInfo.java:187]: There is a change in Class name hierarcy for OzoneAclInfo from OzoneManagerStorageProtos --> OzoneManagerProtocolProtos
and Om.StorageProtocol.proto used for content saved in rocksdb for encode/decode.
Please check if this have any `compatibility issue`, i.e. before upgrade, having encoded with old way in db, and after upgrade, it should able to retrive same information from db with the change.

**Review by sumitagrawl — COMMENTED** [review]
@ivandika3 Thanks for working over this, please check for compatibility case for proto change

File: hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/helpers/OmPrefixInfo.java (hunk 1 above)
Comment (ivandika3) [reply hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/helpers/OmPrefixInfo.java:187]: Thank you for the review. I have tested it, and there was no compatibility issue.

Testing steps:
1. Deploy OM using 1.4.0-rc1 (without this patch)
2. Create a prefix (`ozone sh prefix addacl --acls="group:ozone:w" //dev/vol1/test-bucket/prefix/`)
3. Deploy OM using this patch
4. Read the same prefix (`ozone sh prefix getacl o3://dev/vol1/test-bucket/prefix/`)
5. Add another prefix (`ozone sh prefix addacl --acls="user:testuser:w" //dev/vol1/test-bucket/prefix/`)

Comment (sumitagrawl) [reply hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/helpers/OmPrefixInfo.java:187]: can we read both prefix and print if comming properly?

**Comment by adoroszlai** [comment]
> Note: This seems the separation was done as part of [HDDS-4242](https://issues.apache.org/jira/browse/HDDS-4242), under [HDDS-4216](https://issues.apache.org/jira/browse/HDDS-4216) which are supposed to separate RPC and storage proto files for compatibility reasons by introducing interface-storage package. But it seems only PrefixInfo and UserVolumeInfo have been done for this. If these changes are intended, this can be closed and we can instead migrate the other OzoneAcls to use the one in interface-storage.

Yes, separation / duplication was intended.  I guess we should proceed with HDDS-4465 instead of merging them back.

File: hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/helpers/OmPrefixInfo.java (hunk 1 above)
Comment (ivandika3) [reply hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/helpers/OmPrefixInfo.java:187]: Yes, both type of prefixes can be read and printed properly.

**Comment by ivandika3** [comment]
@adoroszlai Thank you for the info. In that case, let me close this first.

I'll take a look at HDDS-4465 for the outcome and see whether we can proceed with the separation.

**Pull request closed without merging**